File: 백준/Silver/1929. 소수 구하기/소수 구하기.kt
import kotlin.math.*
fun main(args: Array<String>) {
    fun prime_number(N : Int) : MutableList<Int> {
        val isPrime = BooleanArray(N +1) {true}

        isPrime[0] = false
        isPrime[1] = false

        for (i in 2..sqrt(N.toDouble()).toInt()) {
            if (!isPrime[i]){
                continue
            }

            for (j in i * i..N step i) {
                isPrime[j] = false
            }
        }
        var result = mutableListOf<Int>()
        for (i in 0..N) {
            if (isPrime[i]) {
                result.add(i)
            }

        }
        return result
    }
    var (a,b) = readLine()!!.split(" ").map { it.toInt()}
    var primeList = prime_number(b)
    for (i in primeList) {
        if (i >= a) {
            println(i)
        }
    }
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
def prime_numbers(n):
    arr = [i for i in range(n+1)]
    end = int(n**(1/2))
    for i in range(2, end+1):
        if arr[i] == 0:
            continue
        for j in range(i*i, n+1, i):
            arr[j] = 0
    return [i for i in arr[2:] if arr[i]]

a,b = map(int, input().split(" "))    
primeList = prime_numbers(b)
for i in primeList:
    if a <= i:
        print(i)